### PR TITLE
Make language parameter optional

### PR DESCRIPTION
**Pull request by coni2k**
I wanted to try whether I could make language parameter optional, so it can just list all the repos.

I didn't try it with a large sample size but it looks like it's doing the job now.
So, I wanted to share it but feel free to discard it (or make necessary changes) of course.

### CODE REVIEW

File: criticality_score/generate.py [hunk 1]
```diff
@@ -111,7 +110,16 @@ def main():
     repo_urls = set()
     for rnd in range(1, 4):
         print(f'Finding repos (round {rnd}):')
-        repo_urls.update(get_repo_urls(args.language, args.sample_size))
+        urls = []
+        if (args.language):
```
Comment (inferno-chromium) [inline criticality_score/generate.py:114]: Can you move lines 114-120 into a helper function get_github_repo_urls(languages), will be adding gitlab in future, so moving this into a helper is better.

File: criticality_score/generate.py [hunk 2]
```diff
@@ -36,48 +36,47 @@
 }
 IGNORED_KEYWORDS = ['book', 'course', 'docs', 'interview', 'tutorial']
 
-
-def get_repo_urls(languages, sample_size):
+def get_repo_urls(urls, sample_size, github_lang = ''):
```
Comment (inferno-chromium) [inline criticality_score/generate.py:39]: s/get_repo_urls/get_github_repo_urls_for_language

and github_lang=None ('' is not needed).

**Review by inferno-chromium — CHANGES_REQUESTED** [review]
Thanks, some minor changes needed to make future refactoring with gitlab repo easier.

File: criticality_score/generate.py (hunk 1 above)
Comment (coni2k) [reply criticality_score/generate.py]: Okay, it's now like this:

```
repo_urls.update(get_github_repo_urls(args.sample_size, args.language))

def get_github_repo_urls(sample_size, languages):
    urls = []
    if (languages):
        for lang in languages:
            lang = lang.lower()
            for github_lang in LANGUAGE_SEARCH_MAP.get(lang, lang):
                urls = get_github_repo_urls_for_language(urls, sample_size, github_lang)
    else:
        urls = get_github_repo_urls_for_language(urls, sample_size)

    return urls
```

File: criticality_score/generate.py (hunk 2 above)
Comment (coni2k) [reply criticality_score/generate.py]: This one is also updated

```
def get_github_repo_urls_for_language(urls, sample_size, github_lang=None):
```

**Review by inferno-chromium — APPROVED** [review]

**Pull request merged**